scrapinghub/client.py

```diff
@@ -314,27 +770,78 @@ def _modify_iter_filters(self, params):
 
 
 class Items(_Proxy):
+    """Representation of collection of job items.
+
+    Use Job instance to get access to Items.
+
+    Example to retrieve all scraped items from a job:
+        >>> for item in job.items.iter():
+        ...     # do something with item (it's just a dict)
+    """
 
     def _modify_iter_filters(self, params):
+        """Modify iter filter to convert offset to start parameter.
+
+        Returns:
+            dict: updated set of params
+        """
         offset = params.pop('offset', None)
         if offset:
             params['start'] = '{}/{}'.format(self.key, offset)
         return params
 
 
 class Requests(_Proxy):
-
+    """Representation of collection of job requests.
+
+    Use Job instance to get access to Requests.
+
+    Example: retrieving all requests from a job:
+
+        >>> for reqitem in job.requests.iter():
+        ...     # reqitem is a dict
+        >>> reqitem
+        [{
+        'duration': 354,
+        'fp': '6d748741a927b10454c83ac285b002cd239964ea',
+        'method': 'GET',
+        'rs': 1270,
+        'status': 200,a
+        'time': 1482233733870,
+        'url': 'https://example.com'
+        }]
+    """
     def __init__(self, *args, **kwargs):
         super(Requests, self).__init__(*args, **kwargs)
         self._proxy_methods(['add'])
 
 
 class Samples(_Proxy):
-    pass
+    """Representation of collection of job samples.
+
+    Use Job instance to get access to Samples.
+
+    Example: retrieving all samples from a job:
+
+        >>> for sample in j.samples.iter():
+        ...     # sample is a list with timestamp and data
```

Comment:
j -> job, reader shouldn't guess what is j

it'd be nice to show that iter() returns generator object, and that it can take some input parameters, like startts:

```rst
Retrieving all samples from a job::
    
    >>> job.samples.iter()
    <generator object mpdecode at 0x10f5f3aa0>

Retrieving samples with timestamp greater or equal to given timestamp::

    >>> list(job.samples.iter(startts=1484570043851))
    [[1484570043851, 554, 576, 1777, 821, 0], [1484570046673, 561, 583, 1782, 821, 0]]
```

^ the same is valid for other alike objects